provision/internal/gardener/gardener.go

```diff
@@ -89,11 +92,11 @@ func (g *GardenerProvisioner) Credentials(cluster *types.Cluster, provider *type
 	}
 
 	s, err := k8s.CoreV1().Secrets(fmt.Sprintf("garden-%s", provider.ProjectName)).Get(context.Background(), fmt.Sprintf("%s.kubeconfig", cluster.Name), metav1.GetOptions{})
-	if err != nil {
-		return nil, err
+	if err == nil {
+		return s.Data["kubeconfig"], nil
 	}
 
-	return s.Data["kubeconfig"], nil
+	return fetchAdminKubeConfigSubResource(k8s, provider.ProjectName, cluster.Name)
```

Comment:
Since request the `adminkubeconfig` will be the official way to get kubeconfig, it's better we change `Credentials()` function to try this first, then fallback to the secret way, and also add a todo comment to remove the get secret later after gardener kubernetes dropped support for 1.25